src/code42cli/cmds/auditlogs.py

```diff
@@ -153,43 +204,81 @@ def send_to(
         affected_user_ids=affected_user_id,
         affected_usernames=affected_username,
     )
+    if use_checkpoint:
+        checkpoint_name = use_checkpoint
+        events = list(
+            _dedupe_checkpointed_events_and_store_updated_checkpoint(
+                cursor, checkpoint_name, events
+            )
+        )
+    with warn_interrupt():
+        event = None
+        for event in events:
+            logger.info(event)
+        if event is None:  # generator was empty
+            click.echo("No results found.")
 
 
-def _search(sdk, format, **filter_args):
-
-    formatter = OutputFormatter(format, AUDIT_LOGS_DEFAULT_HEADER)
+def _get_all_audit_log_events(sdk, **filter_args):
     response_gen = sdk.auditlogs.get_all(**filter_args)
-
     events = []
     try:
-        for response in response_gen:
-            response_dict = json.loads(response.text)
-            if EVENT_KEY in response_dict:
-                events.extend(response_dict.get(EVENT_KEY))
+        responses = list(response_gen)
```

Comment:
@timabrmsn This is giving me pause and making me reconsider my earlier recommendation that we sort here to protect against future changes in sorting behavior.

If I understand this correctly, if someone were to set up a recurring job with the start date being say, 90 days, ago and checking every 5 minutes, this would load the entire result set into memory and then sort it each time. If that date is still set to the same number a year from now, that in memory-list could get really huge and things are going to get significantly slower.

I realize now that the approach taken here is what is necessary in order to guarantee the sort order, but I failed to recognize that at the time I gave this recommendation. Sorry for the back-and-forth on this, but I think it would be better now if we assumed the default order and spoke to the audit log team to make sure about making sure not to change the default / keeping us in the loop if they do.

Reply:
But if they're running a recurring job, they'd be using a checkpoint, no? And that would cause the begin timestamp to be 5 mins ago for each run after the first one. 

So it's only if the initial run has a ton of events that we'd need to worry about memory/cpu time.

And I don't think we're going to hit any real-world performance problems here. I ran a `auditlogs.get_all()` on my test org and got ~550 events, then made a ton of copies of those to see what memory usage looked like. At around 10million events python was using ~4.5GB, and sorting those results took 3-4 seconds.

Reply:
Oh okay, I totally misunderstood the implementation then. I thought that the checkpoint was just determining which events should be printed, not which shouldn't be fetched.

I haven't touched the CLI in a while so I forgot that the checkpoint time would of course override what was input as the begin time in terminal. Carry on.